### PR TITLE
Separate block positioner select data for performance reasons

### PR DESCRIPTION
**Pull request by raffij**
Resurrect https://github.com/madebymany/sir-trevor-js/pull/408

Address issue:
- https://github.com/madebymany/sir-trevor-js/issues/551

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,10 @@
 
 ## 0.7.x
 
+### 0.7.5
+
+* Separate block positioner select data and make it a property of editor instead of block.
```
Comment (callum) [inline CHANGELOG.md]: Could reference the original issue here that details the problem, only because this doesn't explain why this change was made

Comment (raffij) [reply CHANGELOG.md]: Good point.
